src/main/resources/avro/referencemethods.avdl

```diff
@@ -325,4 +396,99 @@ GetSequenceBasesResponse getSequenceBases(
     /** Additional request parameters to restrict the query. */
     GetSequenceBasesRequest request) throws GAException;
 
+/****************  /joins/search  *******************/
+/**
+This request maps to the body of `POST /joins/search`
+as JSON.
+*/
+record SearchJoinsRequest {
+  /**
+  If present, return only `Join`s which belong to this reference set.
+  */
+  union { null, string } referenceSetId = null;
+  
+  /**
+  If present, return only `Join`s which belong to this variant set.
+  */
+  union { null, string } variantSetId = null;
+
+  // TODO: Specify a proper sort of filter object instead of this. We want to be
+  // able to support a list of sequences, but also a single sequence and a
+  // range.
+
+  /**
+  If nonempty, return only `Join`s with at least one `Position` on one of these
+  sequences.
+  */
+  array<string> sequenceIds = [];
+  
+  /**
+  Must not contain more entries than `sequenceIds`.
```

Comment:
Adam, the way we approached the definitions so far are more like proofs in mathematics - which provide more flexibility - rather than an API implementation which creates bounds and stringency.  Implementation-wise we can also make a specific program's implemented-API more stringent - which might be due to resource-constraints - but the schema definitions should be as flexible as possible to match the intended biological questions (especially the large ones).  I feel the current flow from schema_definition -> implemented_program is easier to go with in the long run, and will allow for extensibility when we want to perform large queries across many repositories such as with Beacon or Matchmaker.  There are always distributed algorithms that can adapt to the definition, so I would lean more to keep the definitions as pure as possible.

~p